File: CMYK-to-RGB/Python/cmyk_rgb_all.py
def cmyk_to_rgb(c, m, y, k, maximums = {}):
    """
    The function cmyk_to_rgb() convert CMYK colors in RGB. You can choose the value ranges in the fifth argument.
    How to use :
    cmyk_to_rgb(c: number, m: number, y: number, k: number, maximums = {
        rgb: {
            r: number = 255,
            g: number = 255,
            b: number = 255
        },
        cmyk: {
            c: number = 255,
            m: number = 255,
            y: number = 255,
            k: number = 255
        }    
    })
    """
    maximums["cmyk"] = getattr(maximums, 'cmyk', {
        "c": 255,
        "m": 255,
        "y": 255,
        "k": 255
    })
    
    maximums["rgb"] = getattr(maximums, 'rgb', {
        "r": 255,
        "g": 255,
        "b": 255
    })
    
    maximums["cmyk"]["c"] = getattr(maximums["cmyk"], "c", 255)
    maximums["cmyk"]["m"] = getattr(maximums["cmyk"], "m", 255)
    maximums["cmyk"]["y"] = getattr(maximums["cmyk"], "y", 255)
    maximums["cmyk"]["k"] = getattr(maximums["cmyk"], "k", 255)
    maximums["rgb"]["r"] = getattr(maximums["rgb"], "r", 255)
    maximums["rgb"]["g"] = getattr(maximums["rgb"], "g", 255)
    maximums["rgb"]["b"] = getattr(maximums["rgb"], "b", 255)

    c_ = c / maximums["cmyk"]["c"]
    m_ = m / maximums["cmyk"]["m"]
    y_ = y / maximums["cmyk"]["y"]
    k_ = k / maximums["cmyk"]["k"]

    if c_ < 0 or c_ > 255 or m_ < 0 or m_ > 255 or y_ < 0 or y_ > 255 or k_ < 0 or k_ > 255 :
        return None
    
    r_ = (1 - c_) * (1 - k_)
    g_ = (1 - m_) * (1 - k_)
    b_ = (1 - y_) * (1 - k_)

    rgb = {
        "r": r_ * maximums["rgb"]["r"],
        "g": g_ * maximums["rgb"]["g"],
        "b": b_ * maximums["rgb"]["b"]
    }

    return rgb


c = float(input("Enter the C value : "))
m = float(input("Enter the M value : "))
y = float(input("Enter the Y value : "))
k = float(input("Enter the K value : "))

print(f"\ncmyk({c}, {m}, {y}, {k}) = rgb({round(cmyk_to_rgb(c, m, y, k)["r"], 2)}, {round(cmyk_to_rgb(c, m, y, k)["g"], 2)}, {round(cmyk_to_rgb(c, m, y, k)["b"], 2)})\n\nAppuyez sur Entrée pour fermer cette fenetre...")
input()

File: CMYK-to-RGB/C/cmyk-rgb-all.c
#include <stdio.h>
#include <math.h>
/*
#ifdef _WIN32
    #include <conio.h>

#else
    #include <unistd.h>
    #include <termios.h>

    int getch(void) {
        struct termios oldattr, newattr;
        int ch;
        tcgetattr(STDIN_FILENO, &oldattr);
        newattr = oldattr;
        newattr.c_lflag &= ~(ICANON | ECHO);
        tcsetattr(STDIN_FILENO, TCSANOW, &newattr);
        ch = getchar();
        tcsetattr(STDIN_FILENO, TCSANOW, &oldattr);
        return ch;
    }
#endif
*/
// Structures for maximum CMYK and RGB values
typedef struct {
    double c;
    double m;
    double y;
    double k;
} CMYK_Max;

typedef struct {
    double r;
    double g;
    double b;
} RGB_Max;

typedef struct {
    CMYK_Max cmyk;
    RGB_Max rgb;
} Maximums;

// Structure for the RGB values
typedef struct {
    int r;
    int g;
    int b;
} RGB;

RGB cmyk_to_rgb(double c, double m, double y, double k, Maximums maximums) {
    /**
    The function cmyk_to_rgb() convert CMYK colors into RGB. You can choose the value ranges in the fifth argument.
    How to use :
    RGB cmykToRgb(double c, double m, double y, double k, Maximums maximums = {
        rgb: {
            double r;
            double g;
            double b;
        },
        cmyk: {
            double c;
            double m;
            double y;
            double k;
        }        
    })
    */
    RGB rgb;
    
    double c_ = c / maximums.cmyk.c;
    double m_ = m / maximums.cmyk.m;
    double y_ = y / maximums.cmyk.y;
    double k_ = k / maximums.cmyk.k;
    
    if (c_ < 0 || c_ > 1 || m_ < 0 || m_ > 1 || y_ < 0 || y_ > 1 || k_ < 0 || k_ > 1    ) {
        rgb.r = -1;
        rgb.g = -1;
        rgb.b = -1;
        return rgb;
    }

    double r_ = (1 - c_) * (1 - k_);
    double g_ = (1 - m_) * (1 - k_);
    double b_ = (1 - y_) * (1 - k_);

    int r = round(maximums.rgb.r * r_);
    int g = round(maximums.rgb.g * g_);
    int b = round(maximums.rgb.b * b_);

    rgb.r = r;
    rgb.g = g;
    rgb.b = b;

    return rgb;
}


// Ask the user for the CMYK values, convert them into RGB, and dispay them.
int main(void) {
    int c, m, y, k;
    printf("Enter the C value : ");
    scanf("%d", &c);
    printf("Enter the M value : ");
    scanf("%d", &m);
    printf("Enter the Y value : ");
    scanf("%d", &y);
    printf("Enter the K value : ");
    scanf("%d", &k);
    
    while (getchar() != '\n');

    RGB_Max rgb_max = {
        .r = 255,
        .g = 255,
        .b = 255
    };

    CMYK_Max cmyk_max = {
        .c = 255,
        .m = 255,
        .y = 255,
        .k = 255
    };
    Maximums maximums = {
        .cmyk = cmyk_max,
        .rgb = rgb_max
    };
    
    printf("\ncmyk(%d, %d, %d, %d) = rgb(%d, %d, %d)\n\nAppuyez sur une touche pour fermer cette fenetre...\n", c, m, y, k, cmyk_to_rgb(c, m, y, k, maximums).r, cmyk_to_rgb(c, m, y, k, maximums).g, cmyk_to_rgb(c, m, y, k, maximums).b);
    getchar();
    return 0;
}